clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -227,12 +227,42 @@ public boolean ready(Node node, long now) {
     }
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    @Override
+    public void disconnect(String nodeId) {
+        selector.close(nodeId);
```

Comment:
The advantage of doing it via the `Selector` is that all `disconnections` are handled together. `abortedSends` are for sends that are not even attempted (and hence `Selector` doesn't know about them). I agree with you that this solution is lower risk and simpler though, so let's go with this PR and consider the `Selector` option in a follow-up.